### PR TITLE
update release notes with latest improvements and fixing broken links in readmes

### PR DESCRIPTION
**Pull request by ancasarb**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -3,12 +3,15 @@
 ### Breaking Changes
```
[the remark this answers is not in the document]
Comment (ancasarb) [reply RELEASE_NOTES.md:5]: updated.

File: README.md [hunk 2]
```diff
@@ -262,7 +262,7 @@ assert(data2(1).getDouble(3) == 0.0)
 
 ## Documentation
 
-For more documentation, please see our [documentation](http://mleap-docs.combust.ml), where you can learn to:
+For more documentation, please see our [documentation](https://combust.github.io/mleap-docs/), where you can learn to:
 
 1. Implement custom transformers that will work with Spark, MLeap and Scikit-learn
```
[the remark this answers is not in the document]
Comment (ancasarb) [reply README.md:267]: yep, will do this just before the release, in a PR like this https://github.com/combust/mleap/pull/754/files.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
 ### New Features
+- MathBinaryModel now supports Logit operations
 
 ### Bug Fixes
 - Fix OneHotEncoder Python serialization
 
 ### Improvements
 - Upgrade to scikit-learn 0.22
+- Upgrade to Spark 3.0.2
+- Upgrade Tensorflow version to 2.4.1
```
[the remark this answers is not in the document]
Comment (ancasarb) [reply RELEASE_NOTES.md:16]: sounds good, we can update the release notes after that PR gets merged.

**Comment by ancasarb** [comment]
@voganrc @jsleight updated the breaking changes section, could you please take another look? thank you!

**Review by jsleight — APPROVED** [review]

**Pull request merged**